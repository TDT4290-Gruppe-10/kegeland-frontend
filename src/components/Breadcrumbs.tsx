import {
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  Text,
} from '@chakra-ui/react';

import useBreadcrumbs from '../hooks/useBreadcrumbs';

type BreadcrumbsProps = React.ComponentProps<typeof Breadcrumb>;

const Breadcrumbs: React.FC<BreadcrumbsProps> = (props) => {
  const { crumbs } = useBreadcrumbs();
  if (crumbs.length === 0) return null;
  return (
    <Breadcrumb {...props}>
      <BreadcrumbItem>
        <BreadcrumbLink href="/">
          <Text color="gray.700">Home</Text>
        </BreadcrumbLink>
      </BreadcrumbItem>
      {crumbs.map((crumb) => (
        <BreadcrumbItem key={crumb.match.pattern.path}>
          <BreadcrumbLink
            href={crumb.match.pathname}
            color={crumb.active ? 'primary.600' : 'gray.700'}>
            {crumb.active ? (
              <Text noOfLines={1} fontWeight="medium">
                {crumb.title}
              </Text>
            ) : (
              <Text
                noOfLines={1}
                maxW="40"
                width="full"
                wordBreak="break-all"
                fontWeight="normal">
                {crumb.title}
              </Text>
            )}
          </BreadcrumbLink>
        </BreadcrumbItem>
      ))}
    </Breadcrumb>
  );
};

export default Breadcrumbs;
