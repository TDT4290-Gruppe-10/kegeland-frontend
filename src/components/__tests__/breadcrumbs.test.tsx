import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Breadcrumbs from '../Breadcrumbs';

describe('Test bredcrumb', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Breadcrumbs />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
