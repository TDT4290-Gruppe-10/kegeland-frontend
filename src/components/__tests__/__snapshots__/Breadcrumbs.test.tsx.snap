// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test bredcrumb renders correctly 1`] = `
<nav
  aria-label="breadcrumb"
  className="chakra-breadcrumb css-0"
>
  <ol
    className="chakra-breadcrumb__list css-70qvj9"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="/"
      >
        <p
          className="chakra-text css-735bf1"
        >
          Home
        </p>
      </a>
    </li>
  </ol>
</nav>
`;
